Source/Core/Core/HLE/HLE.cpp

```diff
@@ -185,13 +185,12 @@ bool IsEnabled(int flags)
 
 u32 UnPatch(const std::string& patch_name)
 {
-  auto* patch = std::find_if(std::begin(OSPatches), std::end(OSPatches), [&](const SPatch& patch) {
-    return patch_name == patch.m_szPatchName;
-  });
+  auto* patch = std::find_if(std::begin(OSPatches), std::end(OSPatches),
```

Comment:
[The compiler will unroll the lambda.](https://godbolt.org/g/7iLvjo) Lambdas behave more or less the same as a static function; if it doesn't get stored somewhere where it would become visible outside the current file then the compiler will inline it.

[Using a lambda with `std::function` will force the compiler to instantiate a real object](https://godbolt.org/g/OoVNOy), then you have to have to pay the full overhead which is why templates are generally preferred for functors than `std::function`.